app/controllers/concerns/active_storage/set_disk_blob.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module ActiveStorage
+  module SetDiskBlob
+    extend ActiveSupport::Concern
+
+    included do
+      before_action :verify_request
+      before_action :set_disk_blob
+    end
+
+    private
+
+    def set_disk_blob
+      return unless @blob.nil?
+
+      @blob = ActiveStorage::Blob.find_by(key: decrypted_hash[:key])
+    rescue StandardError, ActiveSupport::MessageVerifier::InvalidSignature
+      head :not_found
+    end
+
+    def encrypted_hash
+      request.path_info[%r{(?<=disk/).*?(?=-)}]
+    end
+
+    def decrypted_hash
+      # rubocop:disable Security/MarshalLoad
+      Marshal.load(Base64.decode64(JSON.parse(Base64.decode64(encrypted_hash)).dig("_rails", "message")))
+      # rubocop:enable Security/MarshalLoad
+    end
+
+    def verify_request
+      ActiveStorage.verifier.valid_message?(secret) || raise_bad_request
```

Comment:
The `verify` method on the verifier will raise for us so no need for the `|| raise` if you use that method